### PR TITLE
SDI-88: ♻️  More typescript conversion

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: docker-compose-test.yml [hunk 1]
```diff
@@ -23,11 +23,9 @@ services:
       - "9191:8080"
 
   check-my-diary-redis:
-    image: 'bitnami/redis:5.0'
+    image: 'redis:6.2'
```
Comment (petergphillips) [inline docker-compose-test.yml:26]: switch to standard redis image as bitnami one doesn't support arm64

File: integration_tests/integration/health.spec.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+context('Healthcheck', () => {
```
Comment (petergphillips) [inline integration_tests/integration/health.spec.ts:1]: copied from typescript template

File: package.json [hunk 3]
```diff
@@ -121,6 +121,7 @@
     "passport-local": "^1.0.0",
     "passport-oauth2": "^1.6.1",
     "pg": "^8.7.3",
+    "prom-client": "^14.0.1",
```
Comment (petergphillips) [inline package.json:124]: needed for prometheus stuff copied from typescript template

File: server/app.js [hunk 4]
```diff
@@ -145,20 +143,8 @@ module.exports = function createApp({ signInService }) {
...
-      }
-      if (!result.healthy) {
-        res.status(503)
-      }
-      res.json(result)
-      return result
-    })
-  })
-  app.get('/ping', (req, res) => res.send('pong'))
+  app.use(setUpHealthChecks())
```
Comment (petergphillips) [inline server/app.js:146]: copied from typescript template

File: server/authentication/auth.ts [hunk 5]
```diff
@@ -10,10 +11,13 @@ passport.serializeUser((user, done) => {
...
 passport.deserializeUser((user, done) => {
   // Not used but required for Passport
-  done(null, user)
+  done(null, user as Express.User)
 })
 
-function init(signInService) {
+export function init(signInService: {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  getUser: (token: string, refreshToken: string, expiresIn: string, username: string) => any
```
Comment (petergphillips) [inline server/authentication/auth.ts:19]: won't need to do this once more is converted

File: server/middleware/setUpHealthChecks.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import express, { Router } from 'express'
+
+import { healthCheck } from '../services/healthCheck'
```
Comment (petergphillips) [inline server/middleware/setUpHealthChecks.ts:3]: copied from typescript template

File: server/services/healthCheck.ts [hunk 7]
```diff
@@ -0,0 +1,86 @@
+import promClient from 'prom-client'
+import { CommonHttpOption } from 'agentkeepalive'
+
+import { dbCheck, serviceCheckFactory } from '../data/healthCheck'
```
Comment (petergphillips) [inline server/services/healthCheck.ts:4]: copied from typescript template

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**